modules/rke2/variables.tf

```
@@ -0,0 +1,127 @@
+variable "nutanix_username" {
+  description = "The username used to authenticate with Prism Central."
+  type        = string
+}
+
+variable "nutanix_password" {
+  description = "The password used to authenticate with Prism Central."
+  type        = string
+}
+
+variable "nutanix_endpoint" {
+  description = "The endpoint URL for Prism Central."
+  type        = string
+}
+
+variable "nutanix_port" {
+  description = "The port to use when connecting to Prism Central."
+  type        = number
+  default     = 9440
+}
+
+variable "nutanix_insecure" {
+  description = "Set to true to disable SSL certificate validation, false by default."
+  type        = bool
+  default     = false
+}
+
+variable "nutanix_cluster" {
+  description = "The name of the Nutanix cluster to deploy to."
+  type        = string
+}
+
+variable "nutanix_subnet" {
+  description = "The name of the subnet to deploy VMs to."
+  type        = string
+}
+
+variable "name" {
+  description = "Name to use for VMs and other resources."
+  type        = string
+  default     = "rke2-server"
+}
+
+variable "unique_suffix" {
+  description = "Whether to add a unique suffix to the name."
+  type        = bool
+  default     = true
+}
+
+variable "server_count" {
+  description = "The number of server VMs to deploy."
+  type        = number
+  default     = 3
+}
+
+variable "agent_count" {
+  description = "The number of agent VMs to deploy."
+  type        = number
+  default     = 6
+}
+
+
+variable "server_memory" {
+  description = "The amount of memory for each server VM in MiB."
+  type        = number
+}
+
+variable "server_cpu" {
+  description = "The number of CPUs for each server VM."
+  type        = number
+}
+
+variable "server_cpu_cores" {
+  description = "The number of CPU cores, per CPU, for each server VM."
+  type        = number
+  default = 1
+}
+
+variable "agent_memory" {
+  description = "The amount of memory for each agent VM in MiB."
+  type        = number
+}
+
+variable "agent_cpu" {
+  description = "The number of CPUs for each agent VM."
+  type        = number
+}
```

```suggestion
variable "agent_sockets" {
  description = "The number of CPUs for each agent VM."
  type        = number
  default    = 1
}
```
Change verbage to match Nutanix verbage. Makes it more clear what setting this maps to. Also default to 1 since most of the time you probably aren't creating a VM with multiple sockets